src/test/app/Ticket_test.cpp

```diff
@@ -0,0 +1,419 @@
+        BEAST_EXPECT(jticket[sfNewFields.fieldName][jss::Account] ==
+            env.master.human());
+        BEAST_EXPECT(jticket[sfNewFields.fieldName][sfTarget.fieldName] ==
+            alice.human());
+        BEAST_EXPECT(jticket[sfNewFields.fieldName][jss::Sequence] == 2);
+
+        // cancel using the target account
+        auto jv = ticket::create (alice);
+        jv[jss::TransactionType] = "TicketCancel";
+        jv[sfTicketID.fieldName] = jticket[sfLedgerIndex.fieldName];
+        env (jv);
+        auto crd = checkTicketMeta (env, true);
+        auto const& jdir = crd[2];
+        BEAST_EXPECT(jdir[sfLedgerIndex.fieldName] ==
+            jticket[sfLedgerIndex.fieldName]);
+        BEAST_EXPECT(jdir[sfFinalFields.fieldName][jss::Account] ==
+            env.master.human());
+        BEAST_EXPECT(jdir[sfFinalFields.fieldName][sfTarget.fieldName] ==
+            alice.human());
+        BEAST_EXPECT(jdir[sfFinalFields.fieldName][jss::Flags] == 0);
+        BEAST_EXPECT(jdir[sfFinalFields.fieldName][sfOwnerNode.fieldName] ==
+            "0000000000000000");
+        BEAST_EXPECT(jdir[sfFinalFields.fieldName][jss::Sequence] == 2);
+    }
+
+    void testTicketWithExpiration ()
+    {
+        testcase ("Create Ticket with Future Expiration");
+
+        using namespace test::jtx;
+        Env env {*this, features (featureTickets)};
+
+        // create and verify
+        uint32_t expire =
+            (env.timeKeeper ().closeTime () + 60s)
+            .time_since_epoch ().count ();
+        env (ticket::create (env.master, expire));
+        auto cr = checkTicketMeta (env);
+        auto const& jacct = cr[0];
+        auto const& jticket = cr[1];
+        BEAST_EXPECT(
+            jacct[sfPreviousFields.fieldName][sfOwnerCount.fieldName] == 0);
+        BEAST_EXPECT(
+            jacct[sfFinalFields.fieldName][sfOwnerCount.fieldName] == 1);
+        BEAST_EXPECT(jticket[sfNewFields.fieldName][jss::Sequence] ==
+            jacct[sfPreviousFields.fieldName][jss::Sequence]);
+        BEAST_EXPECT(
+            jticket[sfNewFields.fieldName][sfExpiration.fieldName] == expire);
+    }
+
+    void testTicketZeroExpiration ()
+    {
+        testcase ("Create Ticket with Zero Expiration");
+
+        using namespace test::jtx;
+        Env env {*this, features (featureTickets)};
+
+        // create and verify
+        env (ticket::create (env.master, 0u), ter (temBAD_EXPIRATION));
+    }
+
+    void testTicketWithPastExpiration ()
+    {
+        testcase ("Create Ticket with Past Expiration");
+
+        using namespace test::jtx;
+        Env env {*this, features (featureTickets)};
+
+        env.timeKeeper ().adjustCloseTime (days {2});
+        env.close ();
+
+        // create and verify
+        uint32_t expire = 60;
+        env (ticket::create (env.master, expire));
+        // in the case of past expiration, we only get
+        // one meta node entry returned
```

Comment:
well, every tx charges a fee...I think the real point here is that a ticket is not created...so we have no CreatedNode entry